### PR TITLE
Feature/sg 1488 yolo nas r integration model

### PR DESCRIPTION
**Pull request by BloodAxe**

### CODE REVIEW

**Review by shaydeci — CHANGES_REQUESTED** [review]

File: src/super_gradients/training/models/detection_models/yolo_nas_r/yolo_nas_r_dfl_head.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+from super_gradients.common.registry import register_detection_module
+from super_gradients.module_interfaces import SupportsReplaceNumClasses
+from super_gradients.modules import ConvBNReLU
+from super_gradients.modules.base_modules import BaseDetectionModule
+from super_gradients.modules.utils import width_multiplier
+
+
+@register_detection_module()
+class YoloNASRDFLHead(BaseDetectionModule, SupportsReplaceNumClasses):
```
Comment (shaydeci) [inline src/super_gradients/training/models/detection_models/yolo_nas_r/yolo_nas_r_dfl_head.py:15]: Just a reminder from our convo: lets use original dfl head code as inheritance or as a module attribute and simply add the conv for the rotation.

**Review by shaydeci — CHANGES_REQUESTED** [review]
Looks great!
Apart from inline comments and ongoing review, some suggestionsfor the notebook:
- Limitations part: maybe its best to put those at the top?
- I didn't see any mentioning of TRT support, maybe I missed it (ctrl+f tensorrt / trt to verify myself).

Also
- Work on albu support - I think this is a valuable feature. Maybe add a section in our Albu suport botebook for rotated boxes too ?

File: src/super_gradients/training/pretrained_models.py [hunk 2]
```diff
@@ -59,6 +59,9 @@
     "yolo_nas_pose_s_coco_pose": "https://sghub.deci.ai/models/yolo_nas_pose_s_coco_pose.pth",
     "yolo_nas_pose_m_coco_pose": "https://sghub.deci.ai/models/yolo_nas_pose_m_coco_pose.pth",
     "yolo_nas_pose_l_coco_pose": "https://sghub.deci.ai/models/yolo_nas_pose_l_coco_pose.pth",
+    "yolo_nas_r_s_dota2": "https://sghub.deci.ai/models/yolo_nas_r_s_dota2.pth",
```
Comment (shaydeci) [inline src/super_gradients/training/pretrained_models.py:62]: Can we also add a unit test verifying mAP for the pretrained checkpoints ?

Comment (BloodAxe) [reply src/super_gradients/training/pretrained_models.py:62]: It would be hard to do in terms of getting DOTA dataset to cloud instances is not straightforward - there are no "ready-to-use" download links to automate dataset deployment process.
We don't have the dataset on tzags in standard place either :/

File: src/super_gradients/training/models/detection_models/yolo_nas_r/yolo_nas_r_dfl_head.py (hunk 1 above)
Comment (BloodAxe) [reply src/super_gradients/training/models/detection_models/yolo_nas_r/yolo_nas_r_dfl_head.py]: Fixed in 133e507faa0c9999ff731039f0b26f73924308a8

File: src/super_gradients/module_interfaces/exportable_obb_detector.py [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+        to output candidates for NMS. This method may filter out predictions based on confidence threshold and
+        it must obey the contract that the number of predictions per image is fixed and equal to
+        value returned by self.get_num_pre_nms_predictions().
+
+        :param predictions: Input predictions from the model itself.
+        The value of this argument is model-specific
+
+        :return: Implementation of this method must return a tuple of two tensors (boxes, scores) with
+        the following semantics:
+        - boxes - [B, N, 4]
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/module_interfaces/exportable_obb_detector.py]: Indeed. Fixed in df7b85979767fce6c7386f4f07e23fe1c4bd9106

File: src/super_gradients/module_interfaces/exportable_obb_detector.py [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+    A dataclass that holds the result of model export.
+    """
+
+    batch_size: int
+    input_image_channels: int
+    input_image_dtype: torch.dtype
+    input_image_shape: Tuple[int, int]
+
+    quantization_mode: Optional[ExportQuantizationMode]
+
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/module_interfaces/exportable_obb_detector.py:108]: TY! Added in 9b66319bdee1867147c588b8e0a48f94852055e6 and in 50d17fd8284a994e3ab59346655eca9ebafd307c

File: src/super_gradients/module_interfaces/exportable_obb_detector.py [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+                DetectionOutputFormatMode.FLAT_FORMAT - Tensor of shape [N, 8], where N is the total number of
+                predictions in the entire batch.
+                Each row will contain [image_index, cx, cy, w, h, r, class confidence, class index] values.
+
+
+        :param num_pre_nms_predictions: (int) Number of predictions to keep before NMS.
+        :return:
+        """
+
+        # Do imports here to avoid raising error of missing onnx_graphsurgeon package if it is not needed.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/module_interfaces/exportable_obb_detector.py:232]: Ok resolving this one and hope at some point we introduce the simplification PR wit the changes we spoke of (the stuff introduce in the openvino PR).

**Review by shaydeci — CHANGES_REQUESTED** [review]
So the only comment I have left is https://github.com/Deci-AI/super-gradients/pull/2001/files#r1615586708

Didn't we say we can make it inherit from AbstractObjectDetectionDecodingModule?
Sorry if Im mixing some of our talks up...

**Comment by BloodAxe** [comment]
> So the only comment I have left is https://github.com/Deci-AI/super-gradients/pull/2001/files#r1615586708
> 
> Didn't we say we can make it inherit from AbstractObjectDetectionDecodingModule? Sorry if Im mixing some of our talks up...

That's correct. I forgot to add that, thanks for noticing. Fixed in fde191e9d01dfbcb0c90888280a043db630b8db5

**Review by shaydeci — APPROVED** [review]
LGTM
